packages/grid/x-data-grid-premium/src/hooks/features/rowGrouping/useGridRowGrouping.tsx

```diff
@@ -258,7 +260,7 @@ export const useGridRowGrouping = (
   >(() => {
     const sanitizedRowGroupingModel = gridRowGroupingSanitizedModelSelector(apiRef);
     const rulesOnLastRowTreeCreation =
-      apiRef.current.unstable_caches.rowGrouping.rulesOnLastRowTreeCreation;
+      apiRef.current.unstable_caches.rowGrouping.rulesOnLastRowTreeCreation || [];
```

Comment:
Should we have behavior changes in this PR?

```suggestion
      apiRef.current.unstable_caches.rowGrouping.rulesOnLastRowTreeCreation;
```

Reply:
It replaces https://github.com/mui/mui-x/pull/6667/files#diff-1250ff9bde7c6785848b32e7acde51bdce6c5e7d4d82a884a287d33e0e4d8350L267 which eslint complains in `https://eslint.org/docs/latest/rules/default-param-last`.

I can ignore it as well

Reply:
As far as I can tell it's functionally equivalent, but I removed it. We can bring it back in a separate PR.